community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/ValuesExtension.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.schema;
+
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.kernel.lifecycle.Lifecycle;
+import org.neo4j.values.storable.Values;
+
+public class ValuesExtension implements Lifecycle
+{
+    private final LifeSupport life = new LifeSupport();
+    private Config configuration;
+
+    ValuesExtension( ValuesKernelExtensionFactory.Dependencies dependencies )
+    {
+        configuration = dependencies.getConfig();
+    }
+
+    @Override
+    public void init()
+    {
+        life.init();
+    }
+
+    @Override
+    public void start()
+    {
+        life.start();
+        Values.setNanoPrecision( configuration.get( GraphDatabaseSettings.default_temporal_nanosecond_precision ) );
```

Comment:
There can be more than one database running in a JVM, and they can have different configurations.